drivers: qspi: added operation timeout

hlord2000 opened the pull request:
Added devicetree assignment of qspi timeout.
Per nrfx v3.2 addition of qspi timeout in config
struct.

Signed-off-by: Kelly Helmut Lord <kellyhlord@gmail.com>

gmarull commented on dts/bindings/flash_controller/nordic,nrf-qspi.yaml:
```diff
@@ -39,3 +39,8 @@ properties:
   pinctrl-0:
     required: true
+
+  timeout:
+    type: int
+    default: 500
+    description: Timeout for QSPI operations in ms.
```
First, this seems to be a software setting, ie, does not belong to DT.

But it also has DT issues:
- Not ussing suffix (-ms)
- Default not justified (random value here)

hlord2000 committed:
drivers: qspi: added operation timeout

Added Kconfig assignment of qspi timeout.
Per nrfx v3.2 addition of qspi timeout in config
struct.

Signed-off-by: Kelly Helmut Lord <kellyhlord@gmail.com>

gmarull commented on drivers/flash/Kconfig.nordic_qspi_nor:
```diff
@@ -50,4 +50,13 @@ config NORDIC_QSPI_NOR_XIP
+	help
+	  The QSPI peripheral operation timeout in milliseconds.
+	  Primarily intended for long running operations such as
+	  flash mass erase. 500 ms is the default value as
+	  set in the nrfx QSPI driver.
```
This is not a good explanation, nrfx should not be referenced here (Zephyr users may not care about nrfx at all). Just give a brief explanation of why 500ms is a reasonable value.

hlord2000 replied:
I have updated this to reflect typical sector erase time specs

de-nordic reviewed (approved):
Can you add commit with Release Notes entry documenting the addition of Kconfig?
